SingularityBase/src/main/java/com/hubspot/singularity/LoadBalancerUpstream.java

```diff
@@ -43,6 +47,10 @@ public String getGroup() {
     return rackId;
   }
 
+  public Optional<String> getTaskId() {
+    return taskId;
+  }
+
   @Override
   public boolean equals(Object o) {
```

Comment:
I didn't update HashCode / Equals since the absence / presence of this new data doesn't really impact the upstream itself; its just a reference we can use to group things by singularity deploy at the load balancing layer later on. 

Since this data is likely persisted in this form without the value present adding HC / Equals would result in a change to those comparisons and likely break stuff.